src/option_parser.cr

```diff
@@ -1,3 +1,30 @@
+# `OptionParser` is a class for command-line options processing. It supports:
+#
+# * Short and long modifier style options (example: `-h`, `--help`)
+# * Command style arguments (example: `crystal build`)
+# * Passing arguments to the flags (example: `-f filename.txt`)
+# * Automatic help message generation
+#
+# Run `crystal` for an example of a CLI built with `OptionParser`.
```

Comment:
Oh, yeah, the CLI [is built with](https://github.com/manastech/crystal/blob/master/src/compiler/crystal/command.cr#L259) `OptionParser` but the help message is custom, so that's only partially true. Maybe we better come with another example (or remove it altogether)?